internal/executor/reporter/event_sender.go

```diff
@@ -43,23 +45,93 @@ func (eventSender *ExecutorApiEventSender) SendEvents(events []EventMessage) err
 		if err != nil {
 			return err
 		}
+		sequence.Events = filterJobRunEvents(sequence.Events)
 		runId, err := armadaevents.ProtoUuidFromUuidString(e.JobRunId)
 		if err != nil {
-			return fmt.Errorf("Failed to convert uuid string %s to uuid because %s", e.JobRunId, err)
+			return errors.Errorf("failed to convert uuid string %s to uuid because %s", e.JobRunId, err)
 		}
 		populateRunId(sequence, runId)
 		sequences = append(sequences, sequence)
 	}
 	sequences = eventutil.CompactEventSequences(sequences)
-	sequences, err := eventutil.LimitSequencesByteSize(sequences, maxMessageSize, true)
+	if len(sequences) <= 0 {
+		return nil
+	}
+	eventLists, err := splitIntoEventListWithByteLimit(sequences, eventSender.maxMessageSize)
 	if err != nil {
 		return err
 	}
 
-	_, err = eventSender.eventClient.ReportEvents(context.Background(), &executorapi.EventList{Events: sequences})
+	for _, eventList := range eventLists {
+		_, err = eventSender.eventClient.ReportEvents(context.Background(), eventList)
+		if err != nil {
+			return err
+		}
+	}
+
 	return err
 }
 
+func splitIntoEventListWithByteLimit(sequences []*armadaevents.EventSequence, maxEventListSizeBytes int) ([]*executorapi.EventList, error) {
```

Comment:
this function seems a bit weird- we limit the sequences by size by then limit them again when we put them into the event list?
I'm guessing you do this because limitSequencesByByteSize won't account for the overhead of the EventList wrapper.  If it were me I'd just limit the sequences to 2Mb (half of the 4MB grpc limit) and assume that the overhead won't add another 2MB.

Long term we should write a generic function that can take a slice of proto objects and split them appropriately, then we can use the same function for EventSequence and EventList